compiler/test/Dialect/Linalg/fuse-attention.mlir

```diff
@@ -160,3 +160,82 @@ transform.sequence failures(propagate) {
+  // CHECK-LABEL: @multiquery_attention_with_prologue_proj
+  // CHECK: scf.for
+  // CHECK:   scf.for
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK-DAG: tensor.expand_shape
+  // CHECK-DAG: linalg.transpose
+  // CHECK-DAG: linalg.transpose
+  // CHECK-DAG: linalg.generic
+  // CHECK-DAG: linalg_ext.softmax
+  // CHECK-DAG: linalg.generic
+  // CHECK:     scf.yield
+  // CHECK:   scf.yield
+  %cst = arith.constant 0xFF800000 : f32
+  %cst_0 = arith.constant 0.000000e+00 : f32
+  %0 = tensor.empty() : tensor<4x512x512xf32>
+  %20 = tensor.empty() : tensor<4x512x64xf32>
+  %broadcasted = linalg.broadcast ins(%arg1 : tensor<512x512xf32>) outs(%0 : tensor<4x512x512xf32>) dimensions = [0]
+  %1 = tensor.empty() : tensor<4x1024x512xf32>
+  %21 = tensor.empty() : tensor<4x1024x64xf32>
+  %2 = linalg.fill ins(%cst_0 : f32) outs(%1 : tensor<4x1024x512xf32>) -> tensor<4x1024x512xf32>
+  %22 = linalg.fill ins(%cst_0 : f32) outs(%21 : tensor<4x1024x64xf32>) -> tensor<4x1024x64xf32>
+  %3 = linalg_ext.batch_matmul ins(%arg0, %broadcasted : tensor<4x1024x512xf32>, tensor<4x512x512xf32>) outs(%2 : tensor<4x1024x512xf32>) layout = "nn"
+  %broadcasted_1 = linalg.broadcast ins(%arg2 : tensor<512x64xf32>) outs(%20 : tensor<4x512x64xf32>) dimensions = [0]
+  %4 = linalg_ext.batch_matmul ins(%arg0, %broadcasted_1 : tensor<4x1024x512xf32>, tensor<4x512x64xf32>) outs(%22 : tensor<4x1024x64xf32>) layout = "nn"
+  %broadcasted_2 = linalg.broadcast ins(%arg3 : tensor<512x64xf32>) outs(%20 : tensor<4x512x64xf32>) dimensions = [0]
+  %5 = linalg_ext.batch_matmul ins(%arg0, %broadcasted_2 : tensor<4x1024x512xf32>, tensor<4x512x64xf32>) outs(%22 : tensor<4x1024x64xf32>) layout = "nn"
+  %expanded = tensor.expand_shape %3 [[0], [1], [2, 3]] : tensor<4x1024x512xf32> into tensor<4x1024x8x64xf32>
+  %6 = tensor.empty() : tensor<4x8x1024x64xf32>
+  %7 = tensor.empty() : tensor<4x64x1024xf32>
+  %transposed = linalg.transpose ins(%expanded : tensor<4x1024x8x64xf32>) outs(%6 : tensor<4x8x1024x64xf32>) permutation = [0, 2, 1, 3]
+  %transposed_5 = linalg.transpose ins(%4 : tensor<4x1024x64xf32>) outs(%7 : tensor<4x64x1024xf32>) permutation = [0, 2, 1]
+  %8 = tensor.empty() : tensor<4x8x1024x1024xf32>
+  %9 = linalg.fill ins(%cst_0 : f32) outs(%8 : tensor<4x8x1024x1024xf32>) -> tensor<4x8x1024x1024xf32>
+  %10 = tensor.empty() : tensor<4x8x1024xf32>
+  %11 = linalg.fill ins(%cst : f32) outs(%10 : tensor<4x8x1024xf32>) -> tensor<4x8x1024xf32>
+  %12 = linalg.fill ins(%cst_0 : f32) outs(%10 : tensor<4x8x1024xf32>) -> tensor<4x8x1024xf32>
+  %13 = linalg.fill ins(%cst_0 : f32) outs(%6 : tensor<4x8x1024x64xf32>) -> tensor<4x8x1024x64xf32>
+  %14 = linalg.generic {indexing_maps = [#map, #map1, #map2], iterator_types = ["parallel", "parallel", "parallel", "reduction", "parallel"] }
```

Comment:
Why you used generic instead of bmm here? but still use bmm above?

Reply:
Because this test is modified from the test above `multihead_attention_with_prologue_proj`. And I need to change the attention part. Using bmm is also OK, I added a broadcast before it.